pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -185,11 +208,27 @@ def on_pretrain_routine_start(self, trainer, pl_module):
         self.__resolve_ckpt_dir(trainer)
         self.save_function = trainer.save_checkpoint
 
+    def on_train_batch_end(self, trainer, pl_module, *args, **kwargs) -> None:
```

Comment:
so what is not so obvious to see here is: what is if this runs on the last training batch and we will also run the on_validation_end? This would be on the same "global step" right? Yet we have this check `self._last_global_step_saved == trainer.global_step` in the should_skip function. So my question is, do we have a special case here where we would need to save two different checkpoints on the same global step?